charts/cinder-csi-plugin/templates/controllerplugin-deployment.yaml

```diff
@@ -1,21 +1,17 @@
-apiVersion: v1
-kind: Service
-metadata:
-  name: {{ include "cinder-csi.name" . }}-controllerplugin
-  labels:
-    {{- include "cinder-csi.controllerplugin.labels" . | nindent 4 }}
-spec:
-  clusterIP: None
----    
-kind: StatefulSet
+kind: Deployment
```

Comment:
change this will make manifests inconsistent to helm chart .. in case we really need fit for HA 
then a open question is whether we need update manifests as well

Reply:
No it's not a hard requirement and we can keep it as is for better backward compatibility, however the only downside is if a user runs a single replica, he can't benefit from rolling deployment strategy, to have another controller pod ready before removing the old one. The sts-controller first terminate the csi controller pod and then recreates it which causes a short period of service unavailability for single replica ones. For `replicas: 3`, it's not much different. In general it's a better practice to have such leader elected controllers as deployments instead of statefulset. [openshift/openstack-cinder-csi-driver-operator](https://github.com/openshift/openstack-cinder-csi-driver-operator) is using a deployment for cinder-csi, also other csi drivers (such as ceph-csi, etc) provide deployment instead of a statefulSet.

I can update the manifests as well, or if you think it might cause some issues for users during the upgrade, I can revert this to StatefulSet.

Reply:
thanks for the detailed info this is really helpful , I think unless we have special reason
we probably can go deployment as others did so and consider HA reason

I think we don't need to be 100% same for manifest / chart but for this kind of difference (Deployment/Statefulset) , we need some strong reason if we want to make them different
so I guess we can update to deployment and put some upgrade notes in case needed?

@ramineni @lingxiankong any comments?

Reply:
@jichenjc thanks. I updated the manifests as well.